### PR TITLE
Add Feedback into Backdoor1 Branch

### PR DESCRIPTION
**Pull request by itstudentacc**
This merge focuses on adding the following feedback to the most up to date code: which is in backdoor1. This branch was created before main was updated.
Fix Server for secure programming implementation
1.	Update requirements list
2.	Validation of messages
3.	Attach signature to messages
4.	Use logging

### CODE REVIEW

File: server/OlafServer.py
Comment (GregoriusB) [inline server/OlafServer.py:1]: So does this mean we are using a predefined public and private key? Overall, nice work tho 👌

**Review by kylej22 — APPROVED** [review]
Looks good to me

**Review by GregoriusB — APPROVED** [review]
Looks good, I will test it later to make sure backdoor 4 is working

File: server/OlafServer.py
Comment (itstudentacc) [reply server/OlafServer.py:1]: don't have to it can generate one if there is not existing public and private key.

**Pull request merged**